File: Studio Classico/src/main/java/es/uco/iw/classes/Piece.java
package es.uco.iw.classes;

import java.util.Date;

public class Piece {
	
	private int id;
	private String title;
	private String author;
	private int year;
	private int duration;
	private PieceType type;
	private User uploader;
	private String scorePath; // ruta del archivo de la partitura (se dice score en inglés)
	private int nDownloads;
	private Date uploadDate;
	
	public int getID() {
		return id;
	}
	
	public String getTitle() {
		return title;
	}
	
	public String getAuthor() {
		return author;
	}
	
	public int getYear() {
		return year;
	}
	
	public int getDuration() {
		return duration;
	}
	
	public PieceType getPieceType() {
		return type;
	}
	
	public User getUploader() {
		return uploader;
	}
	
	public String getScorePath() {
		return scorePath;
	}
	
	public int getNDownloads() {
		return nDownloads;
	}
	
	public Date getUploadDate() {
		return uploadDate;
	}
	
	public void setID(int id) {
		this.id = id;
	}
	
	public void setTitle(String title) {
		this.title = title;
	}
	
	public void setAuthor(String author) {
		this.author = author;
	}
	
	public void setYear(int year) {
		this.year = year;
	}
	
	public void setDuration(int duration) {
		this.duration = duration;
	}
	
	public void setPieceType(PieceType type) {
		this.type = type;
	}
	
	public void setUploader(User uploader) {
		this.uploader = uploader;
	}
	
	public void setScorePath(String scorePath) {
		this.scorePath = scorePath;
	}
	
	public void setNDownloads(int nDownloads) {
		this.nDownloads = nDownloads;
	}
	
	public void setUploadDate(Date uploadDate) {
		this.uploadDate = uploadDate;
	}
}

File: Studio-Classico/app/Models/Piece.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Piece extends Model
{
    use HasFactory;

    protected $table = "pieces";

    protected $fillable = [
        'title',
		'year',
		'duration',
        'type',
        'author',
		'phone',
        'score_path'
    ];

    protected $hidden = [
        'id',
        'n_downloads',
        'n_visits',
        'upload_date',
        'instruments'
    ];

    public function instruments(){
        return $this->hasMany(InstrumentCount::class);
    }

    public function user(){
        return $this->belongsTo(User::class);
    }
    /*private $id;
    private $title;
    private $year;
    private $duration;
    private $type;
    private $author;
    private $user;
    private $scorePath;
    private $nDownloads;
    private $nVisits;
    private $uploadDate;
    private $instruments;

    function __construct($id, $title, $author, $user, $year, $duration, $type, $scorePath, $nDownloads, $nVisits, $uploadDate, $instruments)
    {
        $this -> setID($id);
        $this -> setTitle($title);
        $this -> setAuthor($author);
        $this -> setUser($user);
        $this -> setYear($year);
        $this -> setScorePath($scorePath);
        $this -> setNDownloads($nDownloads);
        $this -> setNVisits($nVisits);
        $this -> setUploadDate($uploadDate);
        $this -> setInstruments($instruments);
    }

    function getID(){
        return $this -> id;
    }

    function getTitle(){
        return $this -> title;
    }

    function getAuthor(){
        return $this -> author;
    }

    function getUser(){
        return $this -> user;
    }

    function getScorePath(){
        return $this -> scorePath;
    }

    function getNDownloads(){
        return $this -> nDownloads;
    }

    function getNVisits(){
        return $this -> nVisits;
    }

    function getUploadDate(){
        return $this -> uploadDate;
    }

    function getInstruments(){
        return $this -> instruments;
    }

    function setID($id){
        $this -> id = $id;
    }

    function setTitle($title) {
		$this -> title = $title;
	}
	
	function setAuthor($author) {
		$this -> author = $author;
	}
	
	function setUser($user) {
		$this -> user = $user;
	}
	
	function setYear($year) {
		$this -> year = $year;
	}
	
	function setDuration($duration) {
		$this -> duration = $duration;
	}
	
	function setType($type) {
		$this -> type = $type;
	}
	
	function setScorePath($scorePath) {
		$this -> scorePath = $scorePath;
	}
	
	function setNVisits($nVisits) {
		$this -> nVisits = $nVisits;
	}
	
	function setNDownloads($nDownloads) {
		$this -> nDownloads = $nDownloads;
	}
	
	function setUploadDate($uploadDate) {
		$this -> uploadDate = $uploadDate;
	}
	
	function setInstruments($instruments){
		$this -> instruments = $instruments;
	}*/
}
